src/make_watertight/Gen.cpp

```diff
@@ -0,0 +1,2117 @@
+#include <iostream>
+#include <iomanip> // for setprecision
+#include <limits> // for min/max values
+#include <assert.h>
+#include <math.h>
+#include <time.h>
+#include <vector>
+
+#include "moab/GeomTopoTool.hpp"
+#include "moab/FileOptions.hpp"
+
+#include "Gen.hpp"
+#include "moab/Skinner.hpp"
+#include "moab/Range.hpp"
+
+
+const char GEOM_SENSE_2_TAG_NAME[] = "GEOM_SENSE_2";
+const char GEOM_SENSE_N_ENTS_TAG_NAME[] = "GEOM_SENSE_N_ENTS";
+const char GEOM_SENSE_N_SENSES_TAG_NAME[] = "GEOM_SENSE_N_SENSES";
+
+void Gen::moab_printer(moab::ErrorCode error_code) {
+  if (error_code == moab::MB_INDEX_OUT_OF_RANGE) {
+    std::cerr << "ERROR: moab::MB_INDEX_OUT_OF_RANGE" << std::endl;
+  }
+  if (error_code == moab::MB_MEMORY_ALLOCATION_FAILED) {
+    std::cerr << "ERROR: moab::MB_MEMORY_ALLOCATION_FAILED" << std::endl;
+  }
+  if (error_code == moab::MB_ENTITY_NOT_FOUND) {
+    std::cerr << "ERROR: moab::MB_ENTITY_NOT_FOUND" << std::endl;
+  }
+  if (error_code == moab::MB_MULTIPLE_ENTITIES_FOUND) {
+    std::cerr << "ERROR: moab::MB_MULTIPLE_ENTITIES_FOUND" << std::endl;
+  }
+  if (error_code == moab::MB_TAG_NOT_FOUND) {
+    std::cerr << "ERROR: moab::MB_TAG_NOT_FOUND" << std::endl;
+  }
+  if (error_code == moab::MB_FILE_DOES_NOT_EXIST) {
+    std::cerr << "ERROR: moab::MB_FILE_DOES_NOT_EXIST" << std::endl;
+  }
+  if (error_code == moab::MB_FILE_WRITE_ERROR) {
+    std::cerr << "ERROR: moab::MB_FILE_WRITE_ERROR" << std::endl;
+  }
+  if (error_code == moab::MB_ALREADY_ALLOCATED) {
+    std::cerr << "ERROR: moab::MB_ALREADY_ALLOCATED" << std::endl;
+  }
+  if (error_code == moab::MB_VARIABLE_DATA_LENGTH) {
+    std::cerr << "ERROR: moab::MB_VARIABLE_DATA_LENGTH" << std::endl;
+  }
+  if (error_code == moab::MB_INVALID_SIZE) {
+    std::cerr << "ERROR: moab::MB_INVALID_SIZE" << std::endl;
+  }
+  if (error_code == moab::MB_UNSUPPORTED_OPERATION) {
+    std::cerr << "ERROR: moab::MB_UNSUPPORTED_OPERATION" << std::endl;
+  }
+  if (error_code == moab::MB_UNHANDLED_OPTION) {
+    std::cerr << "ERROR: moab::MB_UNHANDLED_OPTION" << std::endl;
+  }
+  if (error_code == moab::MB_FAILURE) {
+    std::cerr << "ERROR: moab::MB_FAILURE" << std::endl;
+  }
```

Comment:
Error Handling: It may make sense to print something here if the `error_code` is unrecognized.